app/src/main/java/com/example/marvelapp/data/request/generator/MarvelRequestGenerator.kt

```diff
@@ -0,0 +1,38 @@
+package com.example.marvelapp.data.request.generator
+
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+class MarvelRequestGenerator {
+
+    private val httpClient = OkHttpClient.Builder().addInterceptor { chain ->
+        val defaultRequest = chain.request()
+        val defaulthttpUrl = defaultRequest.url()
+        val httpUrl = defaulthttpUrl.newBuilder()
+                .addQueryParameter(PUBLIC_API_KEY_ARG, PUBLIC_API_KEY_VALUE)
+                .addQueryParameter(PRIVATE_API_KEY_ARG, PRIVATE_API_HASH_VALUE)
+                .addQueryParameter(TS, TS_VALUE)
+                .build()
+        val requestBuilder = defaultRequest.newBuilder().url(httpUrl)
+        chain.proceed(requestBuilder.build())
+    }
+    private val builder = Retrofit.Builder()
+            .baseUrl(BASE_URL)
+            .addConverterFactory(GsonConverterFactory.create())
+
+    fun <S> createService(serviceClass: Class<S>): S {
+        val retrofit = builder.client(httpClient.build()).build()
+        return retrofit.create(serviceClass)
+    }
+
+    companion object {
+        private const val BASE_URL = "http://gateway.marvel.com/v1/public/"
+        private const val PUBLIC_API_KEY_VALUE = "e6f692f1ab44e78947a2d66faf1dd97d"
```

Comment:
put the api key in a buildConfigField on gradle pls

Reply:
Done thx